drivers/clock_control/clock_control_nrf.c

```diff
@@ -497,38 +466,68 @@ void z_nrf_clock_control_lf_on(enum nrf_lfclk_start_mode start_mode)
 	}
 }
 
-/* Note: this function has public linkage, and MUST have this
- * particular name.  The platform architecture itself doesn't care,
- * but there is a test (tests/kernel/arm_irq_vector_table) that needs
- * to find it to it can set it in a custom vector table.  Should
- * probably better abstract that at some point (e.g. query and reset
- * it by pointer at runtime, maybe?) so we don't have this leaky
- * symbol.
- */
-void nrf_power_clock_isr(void *arg);
+static void clock_event_handler(nrfx_clock_evt_type_t event)
+{
+	struct device *dev = DEVICE_GET(clock_nrf);
+
+	switch (event) {
+	case NRFX_CLOCK_EVT_HFCLK_STARTED:
+	{
+		struct nrf_clock_control_sub_data *data =
+				get_sub_data(dev, CLOCK_CONTROL_NRF_TYPE_HFCLK);
+
+		/* Check needed due to anomaly 201:
+		 * HFCLKSTARTED may be generated twice.
+		 */
+		if (GET_STATUS(data->flags) == CLOCK_CONTROL_STATUS_STARTING) {
+			clkstarted_handle(dev, CLOCK_CONTROL_NRF_TYPE_HFCLK);
+		}
+
+		break;
+	}
+	case NRFX_CLOCK_EVT_LFCLK_STARTED:
+		if (IS_ENABLED(
+			CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION)) {
+			z_nrf_clock_calibration_lfclk_started();
+		}
+		clkstarted_handle(dev, CLOCK_CONTROL_NRF_TYPE_LFCLK);
+		break;
+	case NRFX_CLOCK_EVT_CAL_DONE:
```

Comment:
`NRFX_CLOCK_EVT_CAL_DONE` should not be raised when `CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION` is not enabled. We should raise an ASSERT in that case

Reply:
added